packages/@aws-cdk/aws-stepfunctions-tasks/README.md

```diff
@@ -783,6 +787,198 @@ new tasks.EmrModifyInstanceGroupByName(this, 'Task', {
 });
 ```
 
+## EMR on EKS
+
+Step Functions supports Amazon EMR on EKS through the service integration pattern.
+The service integration APIs correspond to Amazon EMR on EKS APIs, but differ in the parameters that are used.
+
+[Read more](https://docs.aws.amazon.com/step-functions/latest/dg/connect-emr-eks.html) about the differences when using these service integrations.
+
+[Setting up](https://docs.aws.amazon.com/emr/latest/EMR-on-EKS-DevelopmentGuide/setting-up.html) the EKS cluster is required.
+
+### Create Virtual Cluster
+
+The [CreateVirtualCluster](https://docs.aws.amazon.com/emr-on-eks/latest/APIReference/API_CreateVirtualCluster.html) API creates a single virtual cluster that's mapped to a single Kubernetes namespace. 
+
+The EKS cluster containing the Kubernetes namespace where the virtual cluster will be mapped can be passed in from the task input.
+
+```ts
+new tasks.EmrContainersEksCreateVirtualCluster(this, 'Create a Virtual Cluster', {
+  eksCluster: tasks.EksClusterInput.fromTaskInput(sfn.TaskInput.fromText('clusterId')),
+});
+```
+
+The EKS cluster can also be passed in directly.
+
+```ts
+import * as sfn from '@aws-cdk/aws-stepfunctions';
+import * as tasks from '@aws-cdk/aws-stepfunctions-tasks';
+import * as eks from '@aws-cdk/aws-eks';
+
+const eksCluster = new eks.Cluster(this, 'EKS cluster', {
+  version: eks.KubernetesVersion.V1_20,
+});
+
+new tasks.EmrContainersEksCreateVirtualCluster(this, 'Create a Virtual Cluster', {
+  eksCluster: tasks.EksClusterInput.fromCluster(eksCluster),
+});
+```
+
+By default, the Kubernetes namespace that a virtual cluster maps to is "default", but a specific namespace within an EKS cluster can be selected.
+
+```ts
+import * as sfn from '@aws-cdk/aws-stepfunctions';
+import * as tasks from '@aws-cdk/aws-stepfunctions-tasks';
+
+new tasks.EmrContainersEksCreateVirtualCluster(this, 'Create a Virtual Cluster', {
+  eksCluster: tasks.EksClusterInput.fromTaskInput(sfn.TaskInput.fromText('clusterId')),
+  eksNamespace: 'specified-namespace',
+});
+```
+
+### Delete Virtual Cluster
+
+The [DeleteVirtualCluster](https://docs.aws.amazon.com/emr-on-eks/latest/APIReference/API_DeleteVirtualCluster.html) API deletes a virtual cluster.
+
+```ts
+import * as sfn from '@aws-cdk/aws-stepfunctions';
+import * as tasks from '@aws-cdk/aws-stepfunctions-tasks';
+
+new tasks.EmrContainersDeleteVirtualCluster(this, 'Delete a Virtual Cluster', {
```

Comment:
Curious why there's `EmrContainersEksCreateVirtualCluster` but then the `Eks` is omitted from `EmrContainersDeleteVirtualCluster` and `EmrContainersStartJobRun`. I _think_ `Eks` should be in the name of all 3 tasks to differentiate from `Emr` tasks?

Reply:
I guess Matthew and Ben choose to include eks in `EmrContainersEksCreateVirtualCluster` because this deals with an eks cluster as we are creating an emr virtual cluster from an eks cluster. They might have thought that having the word containers is enough to differentiate them from EMR tasks

Reply:
All 3 tasks concern things you can do with "Emr on Eks" right? Tbh I don't know much about either product but I feel like they should follow the same naming convention. How do you feel about `EmrContainersEksDeleteVirtualCluster` and `EmrContainersEksStartJobRun`?

Reply:
How about `EmrContainersOnEks...` as a prefix for all? or just `EmrOnEks...`

Reply:
I think we can go with either `EmrOnEks` or `EmrContainersEks`

Reply:
i am okay with either. Am defaulting to you because I imagine you know more about the integration you are creating and the user experience you are providing. Just want to make sure that they are consistent.

Reply:
How about if we remove the eks from `EmrContainersEksCreateVirtualCluster` since we have `emrcontainers` as the service name for our APIs and it will be consistent for our three calls. As `emrcontainers` means emr on eks.
More about this:

https://awscli.amazonaws.com/v2/documentation/api/latest/reference/emr-containers/index.html

Reply:
I don't love it, but yes lets go with that `EmrContainers...` as a prefix for all 3 tasks.